### ISSUE TITLE
[adsb-multi-portal-feeder] Add Aircraft sensors 

### ISSUE THREAD

**Issue by MaxWinterstein** [MaxWinterstein/homeassistant-addons#172]
Created to continue the discussion that is slightly lost here: https://github.com/MaxWinterstein/homeassistant-addons/commit/5be38c6d740f4645bb49719c5497db1b6d72f36c

@plo53 It looks good, will try to get it integrated and tested! :)

**Comment by MaxWinterstein** [comment]
@plo53 first quick feedback - it works! ❤️ 

Currently drafting the release, maybe get it done before the toddler ends his nap 🤞 

a) Can you add some uuid to the sensor? Maybe someone want's to e.g. overwrite the icon or stuff
![image](https://github.com/MaxWinterstein/homeassistant-addons/assets/5927148/14f814fe-e8ae-453f-aaf0-052931766af9)

b) In the mood to add sensors to the other feeders as well? 😉

**Comment by MaxWinterstein** [comment]
Can be tested with just released version `1.27.0` 🥳

**Comment by plo53** [comment]
this sensor did have a unique id

![sensor_ids](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/8deb0bcb-aa14-4487-af2c-1d7c19c01b29)

**Comment by plo53** [comment]
b) Why not. Need first to register feeding to them, and see what can be done depending on openness of their implementation

**Comment by plo53** [comment]
Just installed 1.27.0
Everything working fine

Didn't reproduce any unique_id missing issue

**Comment by plo53** [comment]
About additional sensors for other feeders:

- fr24 : restfull api 8754 already there

- radarbox : I got trapped into the rbfeeder "segmentation fault" issue within x86 platform. Can't investigate further

- planefinder : it's impossible to create a new sharecode, port 30054 isn't exposed by Thom-x image

**Comment by plo53** [comment]
Good news:
- finally reproduced the unique_id missing issue. It appears that the unique_id identifier I defined previously for sensor.adsdfi_status was somewhere confusing. Fixed it by changing it.
- I managed to get a Planefinger sharecode key by installing the standalone native planefinder docker in an other machine.
Thom-x docker missing port 30053 exposed doesn't allow a new user to register to Planefinder nor to get access to the stats page from HA. Will have a talk with him to access the point.
Anyway I can implement 3 planefinder sensors : Mode-AC packets per seconde, Mode-S packets per seconde, Receiver bandwidth (Kb/s)

**Comment by plo53** [comment]
hassio-sensors v1.1.0 released

- Add Opensky and Planefinder feeders
- Fix unique_id issue

**Comment by plo53** [comment]
New sensors for Opensky and Planefinder in release hassio-sensors v1.1.0

**Comment by MaxWinterstein** [comment]
Awesome ❤️ 

Updated along #174

**Comment by yousaf465** [comment]
@plo53 if you could add a max range sensor too?

**Comment by plo53** [comment]
In local feeders, I may have (or not at all)  limited history of aircraft postions.

So depends by what do you mean by max range.

For adsbexchange and adsb.fi, I can retrieved the outlines. From them I can compute the max distance from the receiver location.

For opensky and planefinder, I have at least the aircraft positions, I can compute the farthest over time.

**Comment by yousaf465** [comment]
for example flight radar gives it as max range 

![image](https://github.com/MaxWinterstein/homeassistant-addons/assets/83491212/c8d259cf-126a-4c98-99db-88231ba73a1f)
 
so max range will give us an idea about our antenna placement etc etc over time.

**Comment by plo53** [comment]
yes indeed, this is the max range for one day history

I prototyped a sensor to track the "farthest" aircraft over time:

![2023-07-12 17_47_51-History – Home Assistant](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/13dd0e36-819a-47b8-a668-0c0264e9c9ae)

Then using a Statistics Graph card like bellow, we should get the same Max Range per day as FR24:

![2023-07-12 18_04_59-Overview – Home Assistant](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/206de71e-bb21-4c81-bcd6-11cc49214972)

**Comment by yousaf465** [comment]
> yes indeed, this is the max range for one day history
> 
> I prototyped a sensor to track the "farthest" aircraft over time:
> 
> ![2023-07-12 17_47_51-History – Home Assistant](https://user-images.githubusercontent.com/55441827/253029435-13dd0e36-819a-47b8-a668-0c0264e9c9ae.png)
> 
> Then using a Statistics Graph card like bellow, we should get the same Max Range per day as FR24:
> 
> ![2023-07-12 18_04_59-Overview – Home Assistant](https://user-images.githubusercontent.com/55441827/253033213-206de71e-bb21-4c81-bcd6-11cc49214972.png)
> 
> Let us wait for some days to see ...

yes wanted this sensor, we can use it for any graph, per hour, per day or per week etc. 

I am using this custom history graph, which creates bar graph on it's own 

https://github.com/alexarch21/history-explorer-card

**Comment by plo53** [comment]
Which feeder are you using ?

PS: Alex's History explorer card is awesome !

**Comment by yousaf465** [comment]
> Which feeder are you using ?
> 
> PS: Alex's History explorer card is awesome !

All feeders 

![image](https://github.com/MaxWinterstein/homeassistant-addons/assets/83491212/06848eda-afba-40a6-8957-d66895e69a27)

**Comment by plo53** [comment]
You may add adsb.fi , launched by some adsbexchange main developpers fired or leave after sold out

**Comment by yousaf465** [comment]
> You may add adsb.fi , launched by some adsbexchange main developpers fired or leave after sold out

it is already added. 
![image](https://github.com/MaxWinterstein/homeassistant-addons/assets/83491212/01c3f4dd-7395-442a-908d-0f3a860a8a8a)

**Comment by plo53** [comment]
hassio-sensors v1.1.2-1 released, to be reviewed by Max

**Comment by MaxWinterstein** [comment]
Update released - works fine. Thanks @plo53 !

Would it be possible to add the sensors to the Home Assistand Addon? Not sure how that via supervisor works, only know it from mqtt site. Rel https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/issues/21

So the sensors will be listed at the addon page?

![image](https://github.com/MaxWinterstein/homeassistant-addons/assets/5927148/d57138bd-fd33-4d0c-9d6d-eff9d866066a)

**Comment by yousaf465** [comment]
> Update released - works fine. Thanks @plo53 !
> 
> Would it be possible to add the sensors to the Home Assistand Addon? Not sure how that via supervisor works, only know it from mqtt site. Rel [MaxWinterstein/toogoodtogo-ha-mqtt-bridge#21](https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/issues/21)
> 
> So the sensors will be listed at the addon page?
> 
> ![image](https://user-images.githubusercontent.com/5927148/253884098-d57138bd-fd33-4d0c-9d6d-eff9d866066a.png)

that will simplify it a lot. just click Add to Dashboard and you are done. instead of finding each and every sensor.

**Comment by plo53** [comment]
Yeah sure. Need to figure out how to integrate the sensors list in the addon page

**Comment by plo53** [comment]
I presume the sensor list in the device is a predifined list by HA, that needs to be handle by the addon.

Here ie the version is not found:

![2023-07-17 11_27_49-Settings – Home Assistant](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/6d1b01ab-27a9-4aa6-ad76-6d8c82bfc4f7)

**Comment by yousaf465** [comment]
https://developers.home-assistant.io/docs/creating_component_index/

does this point to in any right direction?

**Comment by plo53** [comment]
This is developping a custom integration defining a device and its sensors logic.

This is outside of the addon itself.

**Comment by MaxWinterstein** [comment]
Wouldn't it be possible to just add device info to the payload, so it will be created and all sensors will be assigned to this device?

**Comment by plo53** [comment]
The hassio REST API doesn't provide any device settings method.

The "ideal" solution would be an associated custom integration, but I am not in favor because then your add-on would no longer be independent.

I will investigate if replacing REST API communication by MQTT can provide much device control

**Comment by MaxWinterstein** [comment]
At least device autodiscovery would work in that case 🤔 

Btw, I saw this error while fiddling around:
```python
Traceback (most recent call last):
  File "/usr/local/share/planefinder/planefinder_max-range.py", line 28, in <module>
    ulat=radians(float( options['MLAT_EXACT_LAT'] ))
ValueError: could not convert string to float: 'HOMEASSISTANT_LATITUDE'
```

I named this a 'magic variable', as it will be replaced by some script:

https://github.com/MaxWinterstein/homeassistant-addons/blob/052e8c35f89071e9c056f504b657dc7b07a4ca09/adsb-multi-portal-feeder/export-env-from-config.sh#L24

No deeper search here as just on coffee break, but i guess sourcing my bash script on top of the sensors might do the job:
https://github.com/MaxWinterstein/homeassistant-addons/blob/main/adsb-multi-portal-feeder/Dockerfile#L25

**Comment by MaxWinterstein** [comment]
If you want, I can inject this via my dockerfile, just like the others.

**Comment by plo53** [comment]
Initialy I selected to use EXACT_LAT/LON (MLAT) to enable more precise range computing, guessing the site HOMEASSISTANT_LAT/LON may be more approximative.
Anyway, I can update my script to retrieve the values from your env variables.
At the moment, to get rid of the traceback, define  EXACT_LAT/LON isn the addon config.

**Comment by plo53** [comment]
PS: I noted it in the Readme. But your are right, especially for feeder that doesn't enable MLAT, this can trouble users.

![2023-07-18 10_43_41-plo53_adsb-hassio-sensors_ Home assistant _ Aircraft sensors for MaxWinterstein_](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/22b7c640-c13c-49ef-a6af-9413c2048b94)

**Comment by plo53** [comment]
SURVEY : I have implemented in my local dev version a one day tracking of aircrafts seen by my receiver.
Let me know if it brings interest ?
![Home Assistant ADS-B Aircraft Tracked](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/c404af58-88d4-4534-9ef7-4fb69b8e7b5d)

**Comment by plo53** [comment]
> > Would it be possible to add the sensors to the Home Assistand Addon? Not sure how that via supervisor works, only know it from mqtt site. Rel [MaxWinterstein/toogoodtogo-ha-mqtt-bridge#21](https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/issues/21)
> > So the sensors will be listed at the addon page?
> that will simplify it a lot. just click Add to Dashboard and you are done. instead of finding each and every sensor.

I am progressing reworking the adsb sensors from hassio to mqtt with device discovery.

![adsb-mqtt](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/dee9bd2d-eca9-4270-85d3-4c866ff8c8db)

![adsb-mqtt-sensors](https://github.com/MaxWinterstein/homeassistant-addons/assets/55441827/f15e4d08-8d03-4139-8dc7-f3f4a894db0c)

Question : how to manage backward compatibility ?
